MekHQ/src/mekhq/gui/dialog/ForceTemplateAssignmentDialog.java

```diff
@@ -236,12 +237,28 @@ private void updateAssignButtonState() {
      */
     private void assignUnitToTemplate() {
         Unit unit = unitList.getSelectedValue();
-        
         currentScenario.removeUnit(unit.getId());
         currentScenario.addUnit(unit.getId(), templateList.getSelectedValue().getForceName());
         unit.setScenarioId(currentScenario.getId());
         MekHQ.triggerEvent(new DeploymentChangedEvent(unit, currentScenario));
-        
+        if (!unit.getTransportedUnits().isEmpty()) {
+            // Prompt the player to also deploy any units transported by this one
+            int optionChoice = JOptionPane.showConfirmDialog(null,
```

Comment:
I feel like I've seen this exact same dialog elsewhere. Good candidate for a worker function if it's identical.

Reply:
Fixed